src/atomate2/common/analysis/defects/jobs.py

```diff
@@ -166,3 +182,183 @@ def get_ccd_documents(
+    sc_mat_ref: NDArray | None = None,
+    structure_from_prv: Callable | None = None,
+) -> dict:
+    """Get the supercell from the previous calculation.
+
+    Parse the previous calculation directory to obtain the supercell transformation.
+
+    Parameters
+    ----------
+    uc_structure : Structure
+        The unit cell structure of the bulk material.
+    prv_calc_dir : Path
+        The directory of the previous calculation.
+    sc_mat : NDArray
+        The supercell matrix. If not None, use this to validate the extracted supercell.
+    structure_from_prv : Callable
+        Function to get the supercell structure from the previous calculation.
+
+    Returns
+    -------
+    Response:
+        Output containing the supercell transformation and the dir_name
+    """
+    sc_structure = structure_from_prv(prv_calc_dir)
+    (sc_mat_prv, _) = get_matched_structure_mapping(
+        uc_struct=uc_structure, sc_struct=sc_structure
+    )
+
+    if sc_mat_ref is not None:
+        latt_ref = Lattice(sc_mat_ref)
+        latt_prv = Lattice(sc_mat_prv)
+        if not (
+            np.allclose(sorted(latt_ref.abc), sorted(latt_prv.abc))
+            and np.allclose(sorted(latt_ref.angles), sorted(latt_prv.angles))
+        ):
+            raise ValueError(
+                "The supercell matrix extracted from the previous calculation "
+                "does not match the the desired supercell shape."
+            )
+    return dict(sc_mat=sc_mat_prv, lattice=Lattice(sc_mat_prv.lattice))
+
+
+@job(name="bulk supercell")
+def bulk_supercell_calculation(
+    uc_structure: Structure,
+    relax_maker: RelaxMaker,
+    sc_mat: NDArray | None = None,
+) -> Response:
+    """Bulk Supercell calculation.
+
+    Perform a bulk supercell calculation.
+
+    Parameters
+    ----------
+    uc_structure : Structure
+        The unit cell structure.
+    relax_maker : RelaxMaker
+        The relax maker to use.
+    sc_mat : NDArray | None
+        The supercell matrix used to construct the simulation cell.
+
+    Returns
+    -------
+    Response:
+        Output a dictionary containing the bulk supercell calculation summary.
+    """
+    logger.info("Running bulk supercell calculation. Running...")
+    sc_mat = get_sc_fromstruct(uc_structure) if sc_mat is None else sc_mat
+    sc_mat = np.array(sc_mat)
+    sc_structure = uc_structure * sc_mat
+    relax_job = relax_maker.make(sc_structure)
+    relax_job.name = "bulk relax"
+    info = {"sc_mat": sc_mat.tolist()}
+    relax_job.update_maker_kwargs(
+        {"_set": {"write_additional_data->info:json": info}}, dict_mod=True
+    )
+    relax_output: TaskDocument = relax_job.output
+    summary_d = dict(
+        uc_structure=uc_structure,
+        sc_entry=relax_output.entry,
+        sc_struct=relax_output.structure,
+        sc_mat=sc_mat.tolist(),
+        dir_name=relax_output.dir_name,
+        uuid=relax_job.uuid,
+    )
+    return Response(output=summary_d, replace=[relax_job])
```

Comment:
Does this definitely work? I thought if a  job returns a replace, the final output will be the replace flow rather than the output in the original response.

So I would have thought you'd need to do:

```suggestion
    flow = Flow([relax_job], output=summary_d)
    return Response(replace= flow)
```

E.g., see https://github.com/materialsproject/atomate2/blob/aa0bb0737eead988d1dc31fa46d0da0f4e61bc8c/src/atomate2/vasp/jobs/elastic.py#L226-L227

Reply:
You're right the outputs are all references.
It's just that they are all meant to be comsumed by other jobs so this error did not affect anything.